bee-runtime/src/resource.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use log::warn;
+
+use std::{
+    any::{type_name, Any},
+    collections::HashMap,
+    ops::Deref,
+    panic::Location,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc, Mutex, Weak,
+    },
+};
+
+static RES_ID: AtomicUsize = AtomicUsize::new(0);
+
+pub struct ResHandle<R> {
```

Comment:
```suggestion
pub struct ResourceHandle<R> {
```

Reply:
@thibault-martinez Do you have any thoughts on this? I'm ambivalent.

Reply:
I don't have a strong opinion. I am usually in favour of not abbreviating unless it gets way too verbose. I don't think that would be the case here.

Reply:
@zesterer I do agree, that `ResHandle` is one of the obvious abbreviations there are in the world, so it's necessarily a matter of understanding, but more guided by the question, what the abbreviation gains us here?

Reply:
Historically abbreviating things made sense due to display limitations, but nowadays?

Reply:
That doesn't mean, I am totally against it. I just wanted to spawn a discussion to gain consensus on when abbreviations make sense (and there are still reason when they do!), and when not :smile:

Reply:
Resolved, changed to `ResourceHandle`.